src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -104,157 +112,167 @@ class GslQuadAdaptiveImpl {
  protected:
   size_t max_intervals_;
   gsl_function integrand_;
-
   std::unique_ptr<gsl_integration_workspace, gsl_integration_workspace_deleter>
       workspace_;
 };
 }  // namespace detail
 
+enum class IntegralType {
+  StandardGaussKronrod,            // gsl_integration_qag()
+  IntegrableSingularitiesPresent,  // gsl_integration_qags()
+  IntegrableSingularitiesKnown,    // gsl_integration_qagp()
+  InfiniteInterval,                // gsl_integration_qagi()
+  UpperBoundaryInfinite,           // gsl_integration_qagiu()
+  LowerBoundaryInfinite            // gsl_integration_qagil()
+};
+
 /*!
  * \ingroup NumericalAlgorithmsGroup
- * \brief A adaptive integration procedure class
+ * \brief A wrapper around the GSL adaptive integration procedures
  *
- * The class builds a numerical integrator with either an absolute error or
- * relative error and the maximum number of subintervals passed into the
- * constructor. For details on the algorithm see the GSL documentation on
- * `gsl_integration`.
+ * All templates take upper bounds to the absolute and relative error;
+ * 'tolerance_abs' and 'tolerance_rel(default = 0.0)', respectively.
+ * For details on the algorithm see the GSL documentation on `gsl_integration`.
  *
  * Here is an example how to use this class:
  * For the function:
  * \snippet Test_GslQuad.cpp integrated_function
  * the integration should look like:
  * \snippet Test_GslQuad.cpp integration_example
  */
-
-enum class IntegralType {
-  StandardGaussKronrod,            // gsl_integration_qag()
-  IntegrableSingularitiesPresent,  // gsl_integration_qags()
-  IntegrableSingularitiesKnown,  // gsl_integration_qagp(), domain given by pts
-  InfiniteInterval,              // gsl_integration_qagi()
-  UpperBoundaryInfinite,         // gsl_integration_qagiu()
-  LowerBoundaryInfinite          // gsl_integration_qagil()
-};
-
 template <IntegralType TheIntegralType, typename... Args>
 class GslQuadAdaptive;
 
+/*!
+ * The template for "StandardGaussKronrod" uses the QAG algorithm to employ an
+ * adaptive Gauss-Kronrod n-points integration rule. Its function takes a
+ * 'lower_bound' and 'upper_bound' to the integration region, an upper bound for
```

Comment:
you call these arguments `lower_boundary` and `upper_boundary` below, please make consistent.

Also use backticks ` for code